docs/reference/api/lightning_adapter.txt

```diff
@@ -0,0 +1,138 @@
+Determined features, such as mid-epoch preemption, simple distributed
+training interface, simple job submission to the Determined cluster, and
+so on.
+
+LightningAdapter is built on top of our :ref:`PyTorchTrial
+<pytorch-trial>` API, which has a built-in training loop that integrates
+with the Determined features. However, it only supports `LightningModule
+<https://pytorch-lightning.readthedocs.io/en/stable/common/lightning_module.html>`_
+(v1.2.0). To migrate your code from the `Trainer
+<https://pytorch-lightning.readthedocs.io/en/stable/common/trainer.html>`_,
+please read more about :ref:`PyTorchTrial <pytorch-trial>` and
+:ref:`experiment-configuration`.
+
+Porting your ``PyTorchLightning`` code is often pretty simple: 1. Bring
+in your ``LightningModule`` and ``LightningDataModule`` and initialize
+them 2. Create a new trial based on ``LightningAdapter`` and initialize
+it. 3. Define the dataloaders.
+
+Here is an example:
+
+.. code:: python
+
+   from determined.pytorch import PyTorchTrialContext, DataLoader
+   from determined.pytorch.lightning import LightningAdapter
+
+   # bring in your LightningModule and optinally LightningDataModule
+   from mnist import LightningMNISTClassifier, MNISTDataModule
+
+
+   class MNISTTrial(LightningAdapter):
+       def __init__(self, context: PyTorchTrialContext) -> None:
+           # instantiate your LightningModule with hyperparameter from the Determined
+           # config file or from the searcher for automatic hyperparameter tuning.
+           lm = LightningMNISTClassifier(lr=context.get_hparam("learning_rate"))
+
+           # instantiate your LightningDataModule and make it distributed training ready.
+           data_dir = f"/tmp/data-rank{context.distributed.get_rank()}"
+           self.dm = MNISTDataModule(context.get_data_config()["url"], data_dir)
+
+           # initialize LightningAdapter.
+           super().__init__(context, lightning_module=lm)
+           self.dm.prepare_data()
+
+       def build_training_data_loader(self) -> DataLoader:
+           self.dm.setup()
+           dl = self.dm.train_dataloader()
+           return DataLoader(
+               dl.dataset, batch_size=dl.batch_size, num_workers=dl.num_workers
+           )
+
+       def build_validation_data_loader(self) -> DataLoader:
+           self.dm.setup()
+           dl = self.dm.val_dataloader()
+           return DataLoader(
+               dl.dataset, batch_size=dl.batch_size, num_workers=dl.num_workers
+           )
+
+.. autoclass:: determined.pytorch.lightning.LightningAdapter
+   :members:
+   :exclude-members: trial_controller_class, trial_context_class, evaluate_full_dataset, train_batch, evaluate_batch, setup_optimizers_schedulers
+   :inherited-members:
+   :member-order: bysource
+   :special-members: __init__
+
+In this approach, the ``LightningModule`` is not paired with the
+``Trainer`` so that there are some methods and hooks that are not
```

Comment:
I might disambiguate our Trainer with PTL trainer by replacing "paired with the ``Trainer``" -> "paired with a PyTorch Lightning ``Trainer``"

Reply:
👍🏼 
Does "__the__ pytorch lightning trainer" sound good? There seems to be only one.

Reply:
Yep, makes sense to me.